### PR TITLE
#284 Fixes to chained calls from mockk for ImagesRepositoryTest and PartiesRepositoryTest

### LINKED ISSUE #194: Use chained calls from mockk
https://mockk.io/#chained-calls

Remove redundant variables by using chained calls.
`listResult` is totally redundant here
```
	val listResult: ListResult = mockk {
		every { items } returns listOf(storageItem)
	}
	every { imageReference.listAll() } returns mockk {
		coEvery { await() } returns listResult
	}
```
Above can be simplified to:
```
	every { imageReference.listAll() } returns mockk {
		coEvery { await().items } returns listOf(storageItem)
	}
```
Goal of this task is to go throughout all test files and simplify then wherever possible

**TODO:**
- [x] ImagesRepositoryTest - use chained calls
- [x] PartiesRepositoryTest - use chained calls
- [x] ImagesViewModelTest - should be parameterized
- [x] #352

### LINKED ISSUE #284: Use chained calls from mockk for ImagesRepositoryTest and  PartiesRepositoryTest
See #194

### PR DESCRIPTION
**Pull request by kmaslowiec**
Regarding: https://github.com/Android-Guild/Picover/pull/315#discussion_r1358142697

### CODE REVIEW

**Comment by MarekMacko** [comment]
@kmaslowiec Please try to use more descriptive names. "fixes after merge" is too general and it that way we can name everything "minor changes" and probably that it's not the way we would like to follow

File: app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt [hunk 1]
```diff
@@ -36,15 +36,13 @@ class ImagesRepositoryTest : ShouldSpec(
...
-						},
-					)
-				}
+				coEvery { imageReference.listAll().await().items } returns listOf(
+					mockk {
+						every { downloadUrl } returns mockk {
+							coEvery { await() } returns uriResult
+						}
+					},
+				)
```
Comment (MarekMacko) [inline app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:42]: Please verify once again. There is one redundant mockk. Please also check other tests

**Review by MarekMacko — CHANGES_REQUESTED** [review]

File: app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt (hunk 1 above)
Comment (kmaslowiec) [reply app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:42]: Updated.

Comment (MarekMacko) [reply app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:42]: What about this? https://github.com/Android-Guild/Picover/pull/315#discussion_r1358134539

Comment (kmaslowiec) [reply app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:42]: Let's keep the conversation here. Done

Comment (MarekMacko) [reply app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:42]: What about `PartiesRepositoryTest`?

Comment (kmaslowiec) [reply app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:42]: Please suggest which one should be more chained in PartiesRepositoryTest as I don't see any.

Comment (kmaslowiec) [reply app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:42]: @MarekMacko Please Tag me so I can see it in the notification.

Comment (MarekMacko) [reply app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:42]: @kmaslowiec One important rule which seems to be not clear. If someone places a suggestion, fix or improvement, then you should verify all files in your PR, not only the line where comment was posted.  
1. chained calls seems to be ok
2. please use approach of given-when-then in every file, added in you PR #315, e.g. line 44 has redundant line, please also check rest of the lines in `PartiesRepositoryTest.kt`

Comment (kmaslowiec) [reply app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:42]: @MarekMacko Thanks for a detailed explanation. I've checked the files and now both seem to follow given-when-then pattern.

Comment (MarekMacko) [reply app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:42]: Nice! So we're ready to go!

**Comment by MarekMacko** [comment]
@kmaslowiec Please mention the issue in the description to close it after the merge

**Comment by kmaslowiec** [comment]
> @kmaslowiec Please mention the issue in the description to close it after the merge

@MarekMacko I selected it here, will do the trick:
image

**Review by MarekMacko — APPROVED** [review]

**Comment by MarekMacko** [comment]
@kmaslowiec #194 should be closed as well

**Pull request merged**